### PR TITLE
fix race leading to 403 on coordinator insert doc

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

*Race detected on chaos tests, where a db server could have momentarily lost shard leadership, when about to drop a follower. Should lead to retry to new leader instead of 403 to customer*

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
  - [X] **chaos tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [X] Backport for 3.9: *https://github.com/arangodb/arangodb/pull/15600*
  - [X] Backport for 3.8: *https://github.com/arangodb/arangodb/pull/15599*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Cluster/FollowerInfo.h [hunk 1]
```diff
@@ -273,8 +273,9 @@ class FollowerInfo {
           << "Shard " << _docColl->name()
           << " is temporarily in read-only mode, since we could not update the "
              "failover candidates in the agency.";
+      return WriteState::FORBIDDEN;
```
Comment (neunhoef) [inline arangod/Cluster/FollowerInfo.h:276]: Wasn't the idea to return `WriteState::UNAVAILABLE` here?

**Review by neunhoef — CHANGES_REQUESTED** [review]
See comment.

File: arangod/Cluster/FollowerInfo.h (hunk 1 above)
Comment (kvahed) [reply arangod/Cluster/FollowerInfo.h]: done, thx

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by jsteemann — APPROVED** [review]
LGTM!

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**